### DISCUSSION TITLE
remote_spawn for stated_base actor?

### DISCUSSION THREAD

**Discussion by xliao-neology** [actor-framework/actor-framework#1833] [Q&A]
Is that possible to do remote_spawn for state_based actor?
I saw there's an example remote_spawn.cpp which is for function_based. If possible, is it possible to include remote_spawn for state_based actor ?

>>>
template <class T>
struct infer_handle_from_class<T, false> {
  // nop; this enables SFINAE
};

**Comment by Neverlord** [comment]
That looks like an oversight, we'll look into it.

**Comment by riemass** [comment]
In the older releases of CAF, state-based actors were required to inherit from an actor base class and overload a specific member function. We deprecated this behavior with the recent release of CAF, in favor of the `actor_from_state` approach that is now described in the manual and examples but didn't provide a suitable API for remote spawning class-based actors. This is why you get the error with SFINAE failing. 
The old API still works, additionally, I've prepared an MR that implements remote spawn for the new approach.

The new API will look similar to the old approach, after merging it would be possible, for example, to add an actor like this:

```
    add_actor_type<aggregator_state, int32_t&>("aggregator");
```
Here, the aggregator state is the struct that implements the actor, and the additional template parameters (there can be multiple) are arguments to the constructor of the struct. The constructor parameters are required because it is hard to guess them and a struct might implement multiple constructors. They should all be listed as `lvalue` references. @Neverlord I'm not sure if we can relax this requirement.
